### PR TITLE
add multi-scale test

### PR DESCRIPTION
**Pull request by jerrywgz**

### CODE REVIEW

**Review by qingqing01 — COMMENTED** [review]
需要简化architectures部分，和单scale的统一~

File: PaddleCV/PaddleDetection/ppdet/data/transform/operators.py [hunk 1]
```diff
@@ -121,6 +121,122 @@ def __call__(self, sample, context=None):
         return sample
 
 
+@register_op
+class Flip_Augment(BaseOperator):
+    def __init__(self):
```
Comment (qingqing01) [inline PaddleCV/PaddleDetection/ppdet/data/transform/operators.py]: 参考NormalizeImage，Permute加个`is_channel_first`参数，flip时依据这个参数来。

File: PaddleCV/PaddleDetection/ppdet/data/transform/operators.py [hunk 2]
```diff
@@ -121,6 +121,122 @@ def __call__(self, sample, context=None):
         return sample
 
 
+@register_op
+class Flip_Augment(BaseOperator):
```
Comment (qingqing01) [inline PaddleCV/PaddleDetection/ppdet/data/transform/operators.py]: Flip_Augment ->  Flip or FlipImage

File: PaddleCV/PaddleDetection/ppdet/data/transform/operators.py [hunk 3]
```diff
@@ -121,6 +121,122 @@ def __call__(self, sample, context=None):
...
+        super(Flip_Augment, self).__init__()
+        #self.enable = enable
+
+    def __call__(self, sample, context=None):
+        im = sample['image']
+        if not isinstance(im, np.ndarray):
+            raise TypeError("{}: image type is not numpy.".format(self))
+        if len(im.shape) != 3:
+            raise ImageError('{}: image is not 3-dimensional.'.format(self))
+        #if self.enable:
```
Comment (qingqing01) [inline PaddleCV/PaddleDetection/ppdet/data/transform/operators.py]: Remove this line

File: PaddleCV/PaddleDetection/ppdet/data/transform/post_map.py [hunk 4]
```diff
@@ -27,7 +27,9 @@ def build_post_map(coarsest_stride=1,
                    is_padding=False,
                    random_shapes=[],
                    multi_scales=[],
-                   use_padded_im_info=False):
+                   use_padded_im_info=False,
+                   enable_multiscale_test=False,
+                   num_scale=1):
```
Comment (qingqing01) [inline PaddleCV/PaddleDetection/ppdet/data/transform/post_map.py:32]: Add comments for new args

File: PaddleCV/PaddleDetection/ppdet/data/transform/operators.py [hunk 5]
```diff
@@ -121,6 +121,122 @@ def __call__(self, sample, context=None):
...
+        sample['flip_image'] = im[:, ::-1, :]
+        return sample
+
+
+@register_op
+class Multiscale_Test_Resize(BaseOperator):
```
Comment (qingqing01) [inline PaddleCV/PaddleDetection/ppdet/data/transform/operators.py]: Multiscale_Test_Resize -> MultiscaleTestResize

File: PaddleCV/PaddleDetection/ppdet/utils/eval_utils.py [hunk 6]
```diff
@@ -84,6 +130,28 @@ def eval_run(exe, compile_program, pyreader, keys, values, cls):
                 k: (np.array(v), v.recursive_sequence_lengths())
                 for k, v in zip(keys, outs)
             }
+            multi_scale_test = getattr(cfg, 'MultiScaleTEST', None)
+            sub_arch = 'Mask' in cfg.architecture and multi_scale_test
```
Comment (qingqing01) [inline PaddleCV/PaddleDetection/ppdet/utils/eval_utils.py]: sub_arch  -> mask_multi_scale_test

File: PaddleCV/PaddleDetection/ppdet/data/transform/operators.py (hunk 2 above)
Comment (jerrywgz) [reply PaddleCV/PaddleDetection/ppdet/data/transform/operators.py]: Done

File: PaddleCV/PaddleDetection/ppdet/data/transform/operators.py (hunk 1 above)
Comment (jerrywgz) [reply PaddleCV/PaddleDetection/ppdet/data/transform/operators.py]: Done，add use_flip in MultiscaleTestResize

File: PaddleCV/PaddleDetection/ppdet/data/transform/operators.py (hunk 3 above)
Comment (jerrywgz) [reply PaddleCV/PaddleDetection/ppdet/data/transform/operators.py]: Removed

File: PaddleCV/PaddleDetection/ppdet/data/transform/operators.py (hunk 5 above)
Comment (jerrywgz) [reply PaddleCV/PaddleDetection/ppdet/data/transform/operators.py]: Done

File: PaddleCV/PaddleDetection/ppdet/data/transform/post_map.py (hunk 4 above)
Comment (jerrywgz) [reply PaddleCV/PaddleDetection/ppdet/data/transform/post_map.py:32]: Done

File: PaddleCV/PaddleDetection/ppdet/utils/eval_utils.py (hunk 6 above)
Comment (jerrywgz) [reply PaddleCV/PaddleDetection/ppdet/utils/eval_utils.py]: Done

**Review by qingqing01 — APPROVED** [review]

**Pull request merged**